File: src/0921-Minimum-Add-to-Make-Parentheses-Valid/0921.py
class Solution:
    def minAddToMakeValid(self, S):
        """
        :type S: str
        :rtype: int
        """
        result, tmp, = 0, 0
        for s in S:
            tmp += 1 if s == '(' else -1
            if tmp == -1:
                result += 1
                tmp += 1

        return result + tmp

if __name__ == "__main__":
    S = ""
    print(Solution().minAddToMakeValid(S))

File: src/0921-Minimum-Add-to-Make-Parentheses-Valid/0921.cpp
#include <iostream>
#include <vector>
using namespace std;

static int x = []() {std::ios::sync_with_stdio(false); cin.tie(0); return 0; }();
class Solution 
{
public:
    int minAddToMakeValid(string S) 
    {
        int right= 0, left = 0;
        for (auto& s : S)
        {
            s == '(' ? ++left: --left;
            if (left == -1) 
            {
                ++right; ++left;
            }
        }
        return right + left;
    }
};
int main()
{
    string S = "";
    cout << Solution().minAddToMakeValid(S);
    return 0;
}